sources/api/shibaken/src/main.rs

```diff
@@ -81,17 +87,24 @@ fn fetch_public_keys_from_imds() -> Result<Vec<String>> {
 
     info!("Fetching list of available public keys from IMDS");
     // Returns a list of available public keys as '0=my-public-key'
-    let public_key_list =
-        get_text_from_imds(&client, IMDS_PUBLIC_KEY_BASE_URI, &imds_session_token)?;
-    debug!("available public keys '{}'", &public_key_list);
+    let public_key_list = if let Some(public_key_list) =
+        fetch_from_imds(&client, IMDS_PUBLIC_KEY_BASE_URI, &imds_session_token)?
+    {
+        debug!("available public keys '{}'", &public_key_list);
+        public_key_list
+    } else {
+        debug!("no available public keys");
+        return Ok(Vec::new());
+    };
```

Comment:
May be this is more rust idiomatic ? You can workout the var names sorry for that.
```suggestion
let resp =  fetch_from_imds(&client, IMDS_PUBLIC_KEY_BASE_URI, &imds_session_token)?;
        match resp {
            None => {
                    debug!("no available public keys");
                   Ok(Vec::new());
            },
            Some(key_list) => {
                        debug!("available public keys '{}'", &public_key_list);
                      public_key_list
            }
        }
    
```

Reply:
`if let / else` feels more idiomatic to me, or at least it's how I'd write it in this case.

Reply:
It's common and shorter to use if/else for short conditionals like this, particularly with `if let Some(x)`.

Reply:
Okay. I was more worried about return, because clippy complains me for that. It also complains about `let public_key_list = if let Some(public_key_list) =
        fetch_from_imds(&client, IMDS_PUBLIC_KEY_BASE_URI, &imds_session_token)?` 
        and asks to remove from If closure block something.

Reply:
I actually had `Some(s)` originally, but it looked a little bizarre having a single character on Line 94. Using `public_key_list` looked a little nicer and seemed to provide more context, but I can change it back if folks prefer.